### ISSUE TITLE
Create syncfile "finishing" without completing

### ISSUE THREAD

**Issue by jencastrodoesstuff** [digidem/mapeo-desktop#431]
**Describe the bug**
Creating a new syncfile which is 100% data collected from Mm reaches 99% then get's stuck on "Finishing"

**Can you do the same thing twice?**
The exact steps to see the problem again.
1. Go to 'create new syncfile'
2. Write file name and click save in dialogue box
3. Watch data numbers and complete percentage climb as expected 
4. Finishing animation loops. (for over 20 min. now)

**Expected behavior**
"Complete" appears after finishing animation"

**Screenshots**
<img width="513" alt="Screen Shot 2020-10-05 at 1 19 23 PM" src="https://user-images.githubusercontent.com/33264095/95111895-2b295e00-070e-11eb-82e2-e56f7bd43cf9.png">

**Desktop:**
 - OS: macOS
 - Version 10.15.6

**Mapeo Version:**
 - Desktop v5.4.0-beta.9

 
**Additional context**
This is a very large and messy database with about 1000 observations. 900 from ECA RCA. Private content.

**Comment by jencastrodoesstuff** [comment]
Closed Md and got this .. will check back on it later to clear the db and check sync bak from the file created. 
<img width="615" alt="Screen Shot 2020-10-05 at 1 28 33 PM" src="https://user-images.githubusercontent.com/33264095/95112489-fec21180-070e-11eb-9e8a-3a2c88c780aa.png">

**Comment by okdistribute** [comment]
Right, this is the database with missing data in it. Did it cancel after 5 minutes? Perhaps I introduced a regression in the way this was handled, will double check this case.
